cdbot/cogs/cyber.py

```diff
@@ -494,6 +448,92 @@ async def on_message(self, message: Message):
                 await message.channel.send(f"{message.author.mention}  |  {response}")
                 break
 
+    @Cog.listener()
+    async def on_ready(self):
+        readmeFile = "cdbot/data/readme.json"
+        with open(readmeFile, "rb") as f:
+            bytes = f.read()
+            readmeHash = hashlib.sha256(bytes).hexdigest()
+
+        fileHash = None
+        testChannel = await self.bot.fetch_channel(DEV_TESTING_CHANNEL_ID)
+        messages = await testChannel.history().flatten()
+        hashMsg = None
+        for msg in messages:
```

Comment:
This is a linear search of a potentially massive channel, it's gonna be sssslllloooowwwwww. Perhaps it would make more sense to use the channel description for developer testing, as that will always be a fairly constant length? Similar to what we do for the maths webhook.
![image](https://user-images.githubusercontent.com/3729048/85213247-a891b400-b353-11ea-82a6-3fdfe48be107.png)

Reply:
I thought the idea was to use the first pinned message to avoid any searching 🤔

Reply:
I'll do the pin message method then

Reply:
> I thought the idea was to use the first pinned message to avoid any searching altogether thinking

What?

Reply:
it would only work if there arent any other pinned messages actually

Reply:
does the dev testing already have a topic or would it just have the hash?

Reply:
Ive got it working where it grabs it from the channel topic

Reply:
> > I thought the idea was to use the first pinned message to avoid any searching altogether thinking
> 
> What?

We don't have persistent disk, so the idea was pin the hash initially, and for subsequent deploys, compare against the hash in the last pinned message, but channel topic also works I suppose

Reply:
Is there a topic existing that I should add it onto or just leave it as the hash?

Reply:
The current topic is empty

Reply:
Ok then this all works fine